app/workers/facilities/pssg_download.rb

```diff
@@ -56,19 +56,31 @@ def extract_polygon(rings)
     end
 
     def insert_or_update_band(facility)
-      if facility.drivetime_bands.exists?(vha_facility_id: @id, name: @band_name)
-        Rails.logger.info "PSSG Band not updated: Facility #{@id}. Band #{@attributes&.dig('Name')}"
-      else
+      drive_time_band = facility.drivetime_bands.find_by(vha_facility_id: @id, name: @band_name)
+
+      if drive_time_band.nil?
         drive_time_band = facility.drivetime_bands.new(vha_facility_id: @id, name: @band_name)
         drive_time_band.unit = 'minutes'
         drive_time_band.min = round_band(@attributes&.dig('FromBreak'))
         drive_time_band.max = round_band(@attributes&.dig('ToBreak'))
-        drive_time_band.name = @band_name
         drive_time_band.polygon = extract_polygon(@rings)
+        drive_time_band.vssc_extract_date = @vssc_extract_date
 
-        Rails.logger.info "PSSG Band successfully saved: #{@band_name}" # temporary logging
         drive_time_band.save
-        facility.save
+        # Rails.logger.info "PSSG Band successfully inserted: #{@band_name}" # temporary logging
+      elsif @vssc_extract_date > drive_time_band.try(:vssc_extract_date)
+        # rubocop:disable Rails/SkipsModelValidations
+        drive_time_band.update_columns(
+          min: round_band(@attributes&.dig('FromBreak')),
+          max: round_band(@attributes&.dig('ToBreak')),
+          polygon: extract_polygon(@rings),
+          vssc_extract_date: @vssc_extract_date,
+          updated_at: Time.zone.now
+        )
+        # rubocop:enable Rails/SkipsModelValidations
+        # Rails.logger.info "PSSG Band successfully updated: #{@band_name}" # temporary logging
```

Comment:
Any particular reason we're hanging on to these comments?

Reply:
I left a few log lines commented out because although they are overkill for logging everyday, I thought they might be useful for someone coming to reason about this code or watch it run in the future. If it wasn't for the implicit `else` on line 82 I would have just deleted them them already. You are a good test case for that as you haven't looked at this code in months, so if you don't see any value in holding on to them, I can let them go.